tests/integration/test_parts_filters.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+from fastapi.testclient import TestClient
+from unittest.mock import patch, MagicMock
+from typing import Generator
+from uuid import uuid4
+
+from app.main import app
+from app.db.models.builder import Keycap
+
+client = TestClient(app)
+
+FAKE_KEYCAPS = [
+    Keycap(id=uuid4(), name="Keycap1", vendor=["Vendor1"], layout=["ANSI"], material="ABS", profile=["Cherry"], availability=True,
+           price=10.9),
+    Keycap(id=uuid4(), name="Keycap2", vendor=["Vendor2"], layout=["ISO","ANSI"], material="PBT", profile=["OEM"], availability=False,
+           price=20.4),
+    Keycap(id=uuid4(), name="Keycap3", vendor=["Vendor1"], layout=["ANSI"], material="PBT", profile=["MAO"], availability=True,
+           price=15.5),
+    Keycap(id=uuid4(), name="Keycap4", vendor=["Vendor3"], layout=["ANSI"], material="ABS", profile=["NSA"], availability=True,
+           price=25.2),
+    Keycap(id=uuid4(), name="Keycap5", vendor=["Vendor2"], layout=["ISO"], material="ABS", profile=["Cherry"], availability=False,
+           price=30.1),
+]
+
+
+@pytest.fixture(autouse=True)
+def mock_db_query() -> Generator[MagicMock, None, None]:
+    with patch('app.db.crud.get_keycaps') as test_get_keycaps:
+        test_get_keycaps.return_value = FAKE_KEYCAPS
+        yield test_get_keycaps
+
+
+def test_keycap_filter_valid_vendor() -> None:
```

Comment:
In this case, integration tests would be very similar, but instead of mocking the db to return exactly what we specify, we fill the db with the fake data, then let the function use the actual db. This tests both the function as well as well as the db methods, hence integration testing as we're testing that our function and the db and data integrate properly.

Reply:
Another improvement, esp for integration tests, is to check the response body and make sure it matches the expected response. A shortcut for now can be just checking the number of items in the response body matches what is expected.